mobly/controllers/android_device_lib/jsonrpc_client_base.py

```diff
@@ -94,15 +93,21 @@ class JsonRpcClientBase(object):
     of communication.
 
     Attributes:
+        host_port: (int) The host port of this RPC client.
+        device_port: (int) The device port of this RPC client. Must be set by
+          the superclass.
```

Comment:
i'm actually not sure what 'Must be set by the superclass' mean here.
Isn't this the superclass? And I don't see a way to set this. Same with app_name..